common/djangoapps/student/management/tests/test_change_eligibility_deadline.py

```diff
@@ -0,0 +1,107 @@
+        """ Test command with invalid arguments """
+        course_id_str = text_type(self.course.id)
+        username = self.enrolled_user.username
+
+        # Incorrect username
+        with self.assertRaises(User.DoesNotExist):
+            call_command('change_eligibility_deadline',
+                         *command_args.format(username='XYZ', course=course_id_str, date='2018-12-30').split(' ')
+                         )
+        # Incorrect course id
+        with self.assertRaises(InvalidKeyError):
+            call_command('change_eligibility_deadline',
+                         *command_args.format(username=username, course='XYZ', date='2018-12-30').split(' ')
+                         )
+        # Student not enrolled
+        with self.assertRaises(CourseEnrollment.DoesNotExist):
+            unenrolled_user = UserFactory.create()
+            call_command('change_eligibility_deadline',
+                         *command_args.format(username=unenrolled_user.username, course=course_id_str,
+                                              date='2018-12-30').split(' ')
+                         )
+        # Date format Invalid
+        with self.assertRaises(ValueError):
+            call_command('change_eligibility_deadline',
+                         *command_args.format(username=username, course=course_id_str, date='30-12-2018').split(' ')
+                         )
+        # Date not provided
+        with self.assertRaises(KeyError):
+            call_command('change_eligibility_deadline',
+                         *command_args.format(username=username, course=course_id_str,).split(' '))
+
+    def test_invalid_date(self):
+        """
+        Tests the command when the date is prior to today
+
+        In case the date given as deadline is prior to today it sets the deadline to
+        default value which is one month from today. It then continues to run the code
+        to change eligibility deadline.
+        """
+        course_key = text_type(self.course.id)
+        username = self.enrolled_user.username
+
+        # Test Date set prior to today
+        with LogCapture(LOGGER_NAME) as logger:
+            call_command(
+                'change_eligibility_deadline',
+                *command_args.format(username=username, course=course_key, date='2000-12-30').split(' ')
+            )
+            logger.check(
+                (LOGGER_NAME, 'WARNING', 'Invalid date or date not provided. Setting deadline to one month from now'),
+                (LOGGER_NAME, 'INFO', 'Successfully updated credit eligibility deadline for {}'.format(username))
+            )
+
+    def test_valid_command_arguments(self):
+        """ Test command with valid arguments """
+        course_key = text_type(self.course.id)
+        username = self.enrolled_user.username
+
+        call_command('change_eligibility_deadline',
+                     *command_args.format(username=username, course=course_key, date='2030-12-30').split(' ')
```

Comment:
Always try to use `timedelta` to make future dates instead of hardcoding, I have seen many cases where a hardcoded date started failing a test after passing that date.

I know this is not the case here but it is always good to use `timedelta`.